src/composables/useGoalData.js

```diff
@@ -0,0 +1,165 @@
+import {
+	computed,
+	inject,
+	ref,
+} from 'vue';
+
+import thankYouPageQuery from '#src/graphql/query/thankYouPage.graphql';
+import { createUserPreferences, updateUserPreferences } from '#src/util/userPreferenceUtils';
+
+import {
+	ID_BASIC_NEEDS,
+	ID_CLIMATE_ACTION,
+	ID_REFUGEE_EQUALITY,
+	ID_SUPPORT_ALL,
+	ID_US_ECONOMIC_EQUALITY,
+	ID_WOMENS_EQUALITY,
+} from '#src/composables/useBadgeData';
+
+const CATEGORY_TAG_MAP = {
+	[ID_BASIC_NEEDS]: ['basic_needs'],
+	[ID_CLIMATE_ACTION]: ['climate_action'],
+	[ID_REFUGEE_EQUALITY]: ['refugee_support'],
+	[ID_US_ECONOMIC_EQUALITY]: ['us_entrepreneurs'],
+	[ID_WOMENS_EQUALITY]: ['women', '#Woman-Owned Business'],
+};
+
+const GOAL_DISPLAY_MAP = {
+	[ID_BASIC_NEEDS]: 'basic needs loans',
+	[ID_CLIMATE_ACTION]: 'eco-friendly loans',
+	[ID_REFUGEE_EQUALITY]: 'refugees',
+	[ID_US_ECONOMIC_EQUALITY]: 'U.S. entrepreneurs',
+	[ID_WOMENS_EQUALITY]: 'women',
+};
+
+function getGoalDisplayName(category) {
+	return GOAL_DISPLAY_MAP[category] || 'loans';
+}
+
+function getCategoriesForTag(tag) {
+	return Object.entries(CATEGORY_TAG_MAP)
+		.filter(([, tags]) => tags.includes(tag))
+		.map(([category]) => category);
+}
+
+/**
+ * Vue composable for loading and managing user goal data
+ *
+ * @param {Array} loans - List of loans to count toward goals
+ * @returns Goal data and utilities
+ */
+export default function useGoalData(loans) {
+	const apollo = inject('apollo');
+
+	const loading = ref(true);
+	const goalState = ref({});
+	const userPreferences = ref(null);
+	const totalLoans = ref(0);
+	const heroTieredAchievements = ref([]);
+
+	async function loadPreferences() {
+		try {
+			const response = await apollo.query({ query: thankYouPageQuery });
+			const prefsData = response.data?.my?.userPreferences || null;
+			totalLoans.value = response.data?.my?.loans?.totalCount ?? 0;
+			heroTieredAchievements.value = response?.data?.userAchievementProgress?.tieredLendingAchievements ?? [];
```

Comment:
When testing locally was `userAchievementProgress` update right after the checkout transaction? I can't test locally currently due to the dev env being down. I ask because `userAchievementProgress` gets updated via a Kafka listener after the transaction, so there is some delay between transaction and progress being updated. The reason we use `fetchPostCheckoutAchievements` in other TY page modules related to badges is that that query doesn't wait for the Kafta events to run, it just checks "what if" the user checked out with the loans for the current user.